#! /bin/sh -e

# DP: get LDSHARED and CCSHARED (like, CC, CXX, CPP, CFLAGS) from 
# DP: the environment

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        #cd ${dir}gcc && autoconf
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        #rm ${dir}gcc/configure
        ;;
    *)
	echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
        exit 1
esac
exit 0

--- Lib/distutils/sysconfig.py~	2003-02-10 15:02:33.000000000 +0100
+++ Lib/distutils/sysconfig.py	2004-02-03 10:53:21.000000000 +0100
@@ -153,6 +153,10 @@
             cc = os.environ['CC']
         if os.environ.has_key('CXX'):
             cxx = os.environ['CXX']
+        if os.environ.has_key('LDSHARED'):
+            ldshared = os.environ['LDSHARED']
+        if os.environ.has_key('CCSHARED'):
+            ccshared = os.environ['CCSHARED']
         if os.environ.has_key('CPP'):
             cpp = os.environ['CPP']
         else:
